src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -126,30 +119,99 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
    * @param thisNode the input data Node object
    * @return a useful node used to construct the Guava Graph
    */
-  private GraphNode protoNodeToGraphNode(Node thisNode) {
-    return GraphNode.create(thisNode.getName(), thisNode.getTokenList(), thisNode.getMetadata());
+  public GraphNode protoNodeToGraphNode(Node thisNode) {
```

Comment:
What were the reasons behind this change? I thought that the previous version should have worked well.

Reply:
It's because the ones in the proto object are immutable so trying to modify them in add token and remove token was not allowed.

Reply:
Thanks for explaining!